pkg/values/value.go

```diff
@@ -95,11 +102,17 @@ func Wrap(v any) (Value, error) {
 		}
 		return NewList(s)
 	case reflect.Struct:
-		return createMapFromStruct(v)
+		return CreateMapFromStruct(v)
 	case reflect.Pointer:
 		if reflect.Indirect(reflect.ValueOf(v)).Kind() == reflect.Struct {
-			return createMapFromStruct(reflect.Indirect(reflect.ValueOf(v)).Interface())
+			return CreateMapFromStruct(reflect.Indirect(reflect.ValueOf(v)).Interface())
 		}
+	case reflect.String:
+		return Wrap(val.Convert(reflect.TypeOf("")).Interface())
+	case reflect.Bool:
+		return Wrap(val.Convert(reflect.TypeOf(true)).Interface())
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64, reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return Wrap(val.Convert(reflect.TypeOf(int64(0))).Interface())
```

Comment:
why does this work? in other places we have an explicit case for all the int type variations

Reply:
Two reasons,

1 wrapping doesn't need to check boundaries (decoding breaks it down).  And in reflection you can do the equivalent of  (Int64)(int_value) for any int value in the same line of code, but without reflection you can't.